ipam/provider/spec.go

```diff
@@ -0,0 +1,18 @@
+package provider
+
+type GuestClusterConfig struct {
+	Name string
+}
+
+type Interface interface {
+	// CreateCluster creates provider config chart deployment for guest cluster
+	// which provider specific operator reconciles on and therefore yields new
+	// guest cluster with given parameters. This function does not wait for
+	// guest cluster to get ready. This only creates provider config for it.
+	CreateCluster(clusterName string) error
+	// DeleteCluster deletes provider config chart deployment for guest
+	// cluster. Provider specific operator reconciles on it and deletes the
+	// guest cluster eventually. This function does not wait for guest cluster
+	// to get deleted but only deletes the chart deployment of provider config.
+	DeleteCluster(clusterName string)
```

Comment:
IMO this method should return an error and we should always handle it somehow so we do not magically ignore information in case shit hits the fan.

Reply:
I can make this return an error, but especially in test clean-up face there's nothing that can be done if deletion fails.

Reply:
We can log the error.